kernel/mod_mgmt/src/lib.rs

```diff
@@ -2173,6 +2282,35 @@ impl CrateNamespace {
                                 } else {
                                     source_sec_name
                                 };
+
+                                // See `cls_macros` for more details.
+                                if source_sec_name == "__THESEUS_CLS_SIZE" {
+                                    let relocation_entry = RelocationEntry::from_elf_relocation(rela_entry);
+                                    write_relocation(
+                                        relocation_entry,
+                                        target_sec_slice,
+                                        target_sec.mapped_pages_offset,
+                                        VirtualAddress::new(usize::MAX).unwrap(),
```

Comment:
I'd say the problem is more with the `write_relocation` function accepting a `VirtualAddress` rather than a `usize`. The problem with reexporting `kernel_config` through `cls` is that `#cls_offset_expr` can't have a dependency on `cls` because then `preemption` won't compile (`preemption` cannot have a dependency on `cls`).

I've changed the code so that we only ever call
```rust
VirtualAddress::new(usize::MAX).unwrap();
```
on `x64`, even if encountering a `__THESEUS_CLS_SIZE` relocation on AArch64. I think that's the most correct solution.

Reply:
I agree with that and have considered storing a raw `usize` in the virtual address field of the section, considering how TLS (and now CLS) works. I never got around to doing it since most of the sections use a regular virtual address, but yeah, we should probably do that. That change will propagate to the `write_relocation` function too.

For now, go ahead and keep it as is, but can you open an issue with a reference to this comment chain so we don't forget about it? I can implement that later.